fog/report/validation/src/lib.rs

```diff
@@ -57,3 +68,92 @@ pub struct FullyValidatedFogPubkey {
     /// transaction larger than this limit if the fog pubkey is used.
     pub pubkey_expiry: u64,
 }
+
+/// Represents a set of fully validated fog public keys
+///
+/// These can be fetched before building a transaction, then the rest of the
+/// transaction building process can happen offline.
+#[derive(Default, Debug)]
+pub struct FullyValidatedFogPubkeys {
```

Comment:
we could also call this class `PrefetchedFogPubkeys`

Reply:
Feels a little odd to have both a type for `FullyValidatedFogPubkey` and `FullyValidatedFogPubkeys`, but I'm OK with it if it works for you.

Reply:
For me, Manager usually has a very specific meaning in C++ code, of being like a "lifetime manager" for objects of one or several specific types, that owns them, and allows other code to get references to those objects by their id or something like this. Usually the manager is a static global variable.

What do you think about calling it like `FogPubkeySet` ?

Reply:
Not sure what the right naming is. The object both fetches new pubkeys and stores them. To me, `FogPubkeySet` implies that it just stores them.

Reply:
i mean, the resolver fetches them

we could refactor the resolver trait and try to make a better division of responsibilities here

Reply:
Manager is a bad smell unless it means something specific. The GOOS book generally would recommend laying out something about the implementation in the name, to distinguish it from a nop testing object providing a facade functionality . What about FullyValidatedBTreeKeychain or just FullyValidatedKeychain if you don't like the goos book style classname